SlowCheetah.VisualStudio/Utilities/PackageUtilities.cs

```diff
@@ -0,0 +1,147 @@
+    {
+        private static readonly List<string> ExcludedExtensions = new List<string>(new string[] { ".htm", ".html", ".cs.vb", ".txt", ".jpg", ".png", ".ico", ".aspx", ".snk", ".dll", ".pdb", ".settings" });
+
+        /// <summary>
+        /// Given a string which contains a list of extensions separated by semi-colons, this function sorts through the list
+        /// throwing out any blank or empty items, and then makes sure that the extensions start with a ".".
+        /// The validated list is returned.
+        /// </summary>
+        /// <param name="items">A list of extensions separated by semi-colans. The items may be prefixed with the ".", but don't have to be.</param>
+        /// <returns>The validated and culled list of extensions. The returned items all begin with a ".".</returns>
+        public static string[] GetExtensionList(string items)
+        {
+            const string extensionPrefix = ".";
+            List<string> list = new List<string>();
+            if (!string.IsNullOrWhiteSpace(items))
+            {
+                string[] extensions = items.Split(';');
+                foreach (string extension in extensions)
+                {
+                    string extensionValue = string.IsNullOrWhiteSpace(extension) ? string.Empty : extension.Trim();
+                    if (extensionValue != string.Empty)
+                    {
+                        if (!extensionValue.StartsWith(extensionPrefix))
+                        {
+                            extensionValue = extensionPrefix + extensionValue;
+                        }
+
+                        list.Add(extensionValue);
+                    }
+                }
+            }
+            return list.ToArray();
+        }
+
+        /// <summary>
+        /// Gets a list of unsupported transformation extensions,
+        /// each starting with a ".".
+        /// </summary>
+        /// <returns>The list of extensions</returns>
+        public static List<string> GetExcludedExtensions()
+        {
+            return ExcludedExtensions;
+        }
+
+        /// <summary>
+        /// Verifies if the extension of the given file is supported
+        /// </summary>
+        /// <param name="filepath">Full path to the file</param>
+        /// <returns>True if the file is supported</returns>
+        public static bool IsExtensionSupportedForFile(string filepath)
+        {
+            if (string.IsNullOrWhiteSpace(filepath)) { throw new ArgumentNullException("filepath"); }
+            if (!File.Exists(filepath))
+            {
+                throw new FileNotFoundException("File not found", filepath);
+            }
+
+            FileInfo fi = new FileInfo(filepath);
+
+            var isExcludedQuery = from extension in GetExcludedExtensions()
+                                  where string.Compare(fi.Extension, extension, StringComparison.OrdinalIgnoreCase) == 0
+                                  select extension;
```

Comment:
I am personally a fan of the fluent LINQ API. As in: `GetExcludedExtensions().Where(ext => string.Equals(fi.Extension, ext, StringComparison.OrdinalIgnoreCase));`
Also, use `string.Equals`, not `Compare`. And even further, what you are looking for is a `.Any(<delegate from above>)` instead of `.Where`